code/go/internal/validator/semantic/validate_data_stream_streams_template_path.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package semantic
+
+import (
+	"io/fs"
+	"path"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/elastic/package-spec/v3/code/go/internal/fspath"
+	"github.com/elastic/package-spec/v3/code/go/pkg/specerrors"
+)
+
+// ValidateStreamTemplates validates that all referenced template_path files exist for data streams
+func ValidateStreamTemplates(fsys fspath.FS) specerrors.ValidationErrors {
+	var errs specerrors.ValidationErrors
+
+	dataStreams, err := listDataStreams(fsys)
+	if err != nil {
+		return specerrors.ValidationErrors{specerrors.NewStructuredError(err, specerrors.UnassignedCode)}
+	}
+
+	for _, dataStream := range dataStreams {
+		manifestPath := path.Join("data_stream", dataStream, "manifest.yml")
+		streamErrs := validateDataStreamManifestTemplates(fsys, manifestPath, dataStream)
+		errs = append(errs, streamErrs...)
+	}
+
+	return errs
+}
+
+func validateDataStreamManifestTemplates(fsys fspath.FS, manifestPath, dataStreamName string) specerrors.ValidationErrors {
+	var errs specerrors.ValidationErrors
+
+	data, err := fs.ReadFile(fsys, manifestPath)
+	if err != nil {
+		return specerrors.ValidationErrors{specerrors.NewStructuredErrorf("file \"%s\" is invalid: failed to read manifest: %w", fsys.Path(manifestPath), err)}
+	}
+
+	var manifest struct {
+		Streams []struct {
+			Input        string `yaml:"input"`
+			TemplatePath string `yaml:"template_path"`
+		} `yaml:"streams"`
+	}
+
+	err = yaml.Unmarshal(data, &manifest)
+	if err != nil {
+		return specerrors.ValidationErrors{specerrors.NewStructuredErrorf("file \"%s\" is invalid: failed to parse manifest: %w", fsys.Path(manifestPath), err)}
+	}
+
+	for _, stream := range manifest.Streams {
+		if stream.TemplatePath == "" {
+			continue // template_path is optional
+		}
+
+		// First check if template_path is a relative path to the agents/stream directory
+		if strings.HasPrefix(stream.TemplatePath, "./agent/stream") {
+			templatePath := path.Join("data_stream", dataStreamName, stream.TemplatePath)
+			if _, err := fs.Stat(fsys, templatePath); err != nil {
+				errs = append(errs, specerrors.NewStructuredErrorf(
+					"file \"%s\" is invalid: stream \"%s\" references template_path \"%s\" but file does not exist",
+					fsys.Path(manifestPath), stream.Input, stream.TemplatePath))
```

Comment:
This error message could follow the same approach as in `ValidateInputPolicyTemplates` and the error could be shown (and wrapped)

```suggestion
				errs = append(errs, specerrors.NewStructuredErrorf(
					"file \"%s\" is invalid: stream \"%s\" references template_path \"%s\": %w",
					fsys.Path(manifestPath), stream.Input, stream.TemplatePath, err))
```